plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -67,6 +79,37 @@ export type KubernetesBuilderReturn = Promise<{
   serviceLocator: KubernetesServiceLocator;
 }>;
 
+const RESOURCE_TYPE_KUBERNETES_RESOURCE = 'kubernetes-resource';
+type K8sQuery = Set<{
+  apiVersion: string;
+  kind: string;
+  cluster?: string;
+  namespace?: string;
+  labelSelector?: string;
+  subResource?: string;
+}>;
+const createKubernetesPermissionRule = makeCreatePermissionRule<
+  object,
+  K8sQuery,
+  typeof RESOURCE_TYPE_KUBERNETES_RESOURCE
+>();
+const matches = createKubernetesPermissionRule({
+  name: 'MATCHES',
+  description: 'Allow kubernetes resources matching the given structure',
+  resourceType: RESOURCE_TYPE_KUBERNETES_RESOURCE,
+  apply: (resource: object, params: PermissionRuleParams) => {
+    try {
+      expect(resource).toMatchObject(params!);
+    } catch {
+      return false;
+    }
+    return true;
```

Comment:
This is obviously pretty hacky, but felt like a nice easy way to illustrate "generalized object matching"